File: data/src/main/java/org/cryptomator/data/cloud/webdav/WebDavFile.java
package org.cryptomator.data.cloud.webdav;

import org.cryptomator.domain.Cloud;
import org.cryptomator.domain.CloudFile;
import org.cryptomator.util.Optional;

import java.util.Date;

public class WebDavFile implements CloudFile, WebDavNode {

	private final WebDavFolder parent;
	private final String name;
	private final String path;
	private final Optional<Long> size;
	private final Optional<Date> modified;

	public WebDavFile(WebDavFolder parent, String name, Optional<Long> size, Optional<Date> modified) {
		this(parent, name, parent.getPath() + "/" + name, size, modified);
	}

	public WebDavFile(WebDavFolder parent, String name, String path, Optional<Long> size, Optional<Date> modified) {
		this.parent = parent;
		this.name = name;
		this.path = path;
		this.size = size;
		this.modified = modified;
	}

	@Override
	public Cloud getCloud() {
		return parent.getCloud();
	}

	@Override
	public String getName() {
		return name;
	}

	@Override
	public String getPath() {
		return path;
	}

	@Override
	public WebDavFolder getParent() {
		return parent;
	}

	@Override
	public Optional<Long> getSize() {
		return size;
	}

	@Override
	public Optional<Date> getModified() {
		return modified;
	}
}

File: data/src/main/java/org/cryptomator/data/cloud/webdav/WebDavFile.kt
package org.cryptomator.data.cloud.webdav

import org.cryptomator.domain.Cloud
import org.cryptomator.domain.CloudFile
import java.util.Date

class WebDavFile(override val parent: WebDavFolder, override val name: String, override val path: String, override val size: Long?, override val modified: Date?) : CloudFile, WebDavNode {

	constructor(parent: WebDavFolder, name: String, size: Long?, modified: Date?) : this(parent, name, parent.path + "/" + name, size, modified)

	override val cloud: Cloud?
		get() = parent.cloud
}
